app/Http/Controllers/CategoryController.php

```diff
@@ -24,28 +24,30 @@ public function store(CategoryRequest $request)
     {
         Category::create($request->all());
 
-        return redirect()->route('categories.index')->with('success', 'Added successfully ');
+        return redirect()->route('categories.index')->with('success', 'Category added successfully ');
     }
-    public function edit(string $catId)
+    public function edit(string $categoryId)
     {
-        $categories = Category::select('id', 'name', 'description', 'parent_id')->get();
-        $category = Category::select('id', 'name', 'description', 'parent_id')->find($catId) ;
+        $categories = Category::select('id', 'name', 'description')->get();
+        $category = Category::select('id', 'name', 'description', 'parent_id')->find($categoryId);
         
         return view('admin.pages.category.form', compact('category','categories'));
     }
     public function update(CategoryRequest $request, Category $category)
     {
         $category->update($request->all());
 
-        return redirect()->route('categories.index')->with('success', 'Category Updated successfully.');
+        return redirect()->route('categories.index')->with('success', 'Category updated successfully.');
     }
 
-    public function destroy($id)
+    public function destroy(string $categoryId)
     {
-        $category = Category::find($id);
+         $category = Category::select('id')->find($categoryId);
 
+        if ($category->children->isNotEmpty()) {
+            return redirect()->route('categories.index')->with('error', 'Cannot delete category with child categories.');
+        }
```

Comment:
query is not proper.

message is: `This category cannot be deleted as there are one or more child categories attached.`